### PR TITLE
patch: Track test skips & summarize the summaries

### PR DESCRIPTION
**Pull request by vipulgupta2048**
Improvements made: 
1. Merge all summaries into one summary JSON file
2. Add skipped stats and summary creation time. 
3. UX: Deleting individual summary files to give a cleaner look in Jenkins

```
[
  {
    "suite": "Testbot Diagnositcs",
    "stats": {
      "tests": 4,
      "ran": 3,
      "skipped": 1,
      "passed": 3,
      "failed": 0
    },
    "tests": {
      "Flashing an image": "passed",
      "Power cycling the DUT": "skipped",
      "Is the DUT reachable?": "passed",
      "Recording DUT serial output": "passed"
    },
    "dateTime": "Mon Feb 21 2022 20:38:24 GMT+0000 (Coordinated Universal Time)"
  },
  {
    "suite": "Unmanaged BalenaOS release suite",
    "stats": {
      "tests": 3,
      "ran": 3,
      "skipped": 0,
      "passed": 3,
      "failed": 0
    },
    "tests": {
      "resinos.fingerprint file test": "passed",
      "ext4 filesystems are checked on boot": "passed",
      "OS-release file check": "passed"
    },
    "dateTime": "Mon Feb 21 2022 20:45:02 GMT+0000 (Coordinated Universal Time)"
  }
]
```

Signed-off-by: Vipul Gupta (@vipulgupta2048) <vipul@balena.io>

### CODE REVIEW

**Comment by vipulgupta2048** [comment]
@balena-ci rebase

File: client/bin/multi-client.js [hunk 1]
```diff
@@ -232,6 +232,15 @@ class NonInteractiveState {
...
 		);
+
+		let summaries = [];
+		nativeFs.readdirSync(`reports`).forEach((file) => {
+			if (/^test-summary-(.*).json$/.test(file)) {
+				summaries.push(fs.readFileSync('reports/' + file).toString());
+				fs.unlinkSync('reports/' + file);
+			}
+		});
+		nativeFs.writeFileSync(`reports/final-result`, summaries.join(',\n'));
```
Comment (klutchell) [inline client/bin/multi-client.js:243]:
```suggestion
		nativeFs.writeFileSync(`reports/final-result`, JSON.stringify(summaries));
```
Wouldn't something like this output valid JSON (an array of summary objects)?

File: client/bin/multi-client.js [hunk 2]
```diff
@@ -371,82 +380,82 @@ class NonInteractiveState {
...
+							'-s',
+							job.suite,
+							'-u',
+							job.workers,
+						],
+						{
+							stdio: 'pipe',
+							env: {
+								CI: true,
+							},
```
Comment (klutchell) [inline client/bin/multi-client.js:411]: Can you move all these linting changes into a separate commit?

File: core/lib/common/suite.js [hunk 3]
```diff
@@ -129,13 +129,16 @@ class Suite {
...
 		this.testSummary = {
 			suite: ``,
 			stats: {
 				tests: 0,
-				passes: 0,
-				fails: 0,
 				ran: 0,
+				skipped: () => this.testSummary.stats.tests - this.testSummary.stats.ran,
+				passed: 0,
+				failed: 0,
```
Comment (klutchell) [inline core/lib/common/suite.js:141]: :+1:

File: client/bin/multi-client.js (hunk 1 above)
Comment (vipulgupta2048) [reply client/bin/multi-client.js:243]: No, because the array part is what is missing from our final-result. 

We got the objects perfectly formed and formatted, I am also adding a nice `,` there too but the capital brackets that form the actual array isn't there. I can manually concatenate them if you like? What do you think?

File: client/bin/multi-client.js (hunk 2 above)
Comment (vipulgupta2048) [reply client/bin/multi-client.js]: Done, and probably it was a false negative

File: client/bin/multi-client.js (hunk 1 above)
Comment (vipulgupta2048) [reply client/bin/multi-client.js:243]: Oh you made a suggestion, I think I tried this and let me re-run and paste the output of this.

Comment (vipulgupta2048) [reply client/bin/multi-client.js:243]:
```
["{\n    \"suite\": \"Testbot Diagnositcs\",\n    \"stats\": {\n        \"tests\": 4,\n        \"ran\": 3,\n        \"skipped\": 1,\n        \"passed\": 3,\n        \"failed\": 0\n    },\n    \"tests\": {\n        \"Flashing an image\": \"passed\",\n        \"Power cycling the DUT\": \"skipped\",\n        \"Is the DUT reachable?\": \"passed\",\n        \"Recording DUT serial output\": \"passed\"\n    },\n    \"dateTime\": \"Mon Feb 21 2022 13:47:56 GMT+0000 (Coordinated Universal Time)\"\n}","{\n    \"suite\": \"Unmanaged BalenaOS release suite\",\n    \"stats\": {\n        \"tests\": 3,\n        \"ran\": 3,\n        \"skipped\": 0,\n        \"passed\": 3,\n        \"failed\": 0\n    },\n    \"tests\": {\n        \"resinos.fingerprint file test\": \"passed\",\n        \"ext4 filesystems are checked on boot\": \"passed\",\n        \"OS-release file check\": \"passed\"\n    },\n    \"dateTime\": \"Mon Feb 21 2022 13:54:09 GMT+0000 (Coordinated Universal Time)\"\n}"]
```

It's basically an array of 2 strings. I will see if I can JSON.parse  this mess if not this is our best shot

Comment (vipulgupta2048) [reply client/bin/multi-client.js]:
```
[{"suite":"Testbot Diagnositcs","stats":{"tests":4,"ran":3,"skipped":1,"passed":3,"failed":0},"tests":{"Flashing an image":"passed","Power cycling the DUT":"skipped","Is the DUT reachable?":"passed","Recording DUT serial output":"passed"},"dateTime":"Mon Feb 21 2022 19:13:12 GMT+0000 (Coordinated Universal Time)"},{"suite":"Unmanaged BalenaOS release suite","stats":{"tests":3,"ran":3,"skipped":0,"passed":3,"failed":0},"tests":{"resinos.fingerprint file test":"passed","ext4 filesystems are checked on boot":"passed","OS-release file check":"passed"},"dateTime":"Mon Feb 21 2022 19:18:40 GMT+0000 (Coordinated Universal Time)"}]
```

Thanks for making me follow up on this, as json.parse did it.

**Comment by vipulgupta2048** [comment]
@balena-ci rebase

**Comment by vipulgupta2048** [comment]
@balena-ci I self-certify!
